homeassistant/components/sensor/github.py

```diff
@@ -0,0 +1,223 @@
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return self._available
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return {
+            ATTR_PATH: self._repository_path,
+            ATTR_NAME: self._name,
+            ATTR_LAST_COMMIT_MESSAGE: self._last_commit_message,
+            ATTR_LAST_COMMIT_SHA: self._last_commit_sha,
+            ATTR_LATEST_RELEASE_URL: self._latest_release_url,
+            ATTR_OPEN_ISSUE_LATEST: self._open_issue_latest,
+            ATTR_OPEN_ISSUES: self._open_issue_count,
+            ATTR_OPEN_PULL_REQUEST_LATEST: self._pull_request_latest,
+            ATTR_OPEN_PULL_REQUESTS: self._pull_request_count,
+            ATTR_STARGAZERS: self._stargazers
+        }
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend."""
+        return 'mdi:github-circle'
+
+    def update(self):
+        """Collect updated data from GitHub API."""
+        self._github_data.update()
+
+        self._name = self._github_data.name
+        self._state = self._github_data.last_commit_sha
+        self._repository_path = self._github_data.repository_path
+        self._available = self._github_data.available
+        self._last_commit_message = self._github_data.last_commit_message
+        self._last_commit_sha = self._github_data.last_commit_sha
+        self._latest_release_url = self._github_data.latest_release_url
+        self._open_issue_count = self._github_data.open_issue_count
+        self._open_issue_latest = self._github_data.open_issue_latest
+        self._pull_request_count = self._github_data.pull_request_count
+        self._pull_request_latest = self._github_data.pull_request_latest
+        self._stargazers = self._github_data.stargazers
+
+
+class GitHubData():
+    """GitHub Data object."""
+
+    def __init__(self, repository, access_token=None, server_url=None):
+        """Set up GitHub."""
+        import github
+
+        self._github = github
+
+        self.setup_error = False
+
+        try:
+            if server_url is not None:
+                server_url += "/api/v3"
+                if access_token is not None:
+                    self._github_obj = github.Github(
+                        access_token, base_url=server_url)
+            else:
+                if access_token is not None:
+                    self._github_obj = github.Github(access_token)
+
+            self.repository_path = repository[CONF_PATH]
+
+            repo = self._github_obj.get_repo(self.repository_path)
+        except self._github.GithubException as err:
+            _LOGGER.error("GitHub error for %s: %s", self.repository_path, err)
+            self.setup_error = True
+            return
+
+        if CONF_NAME in repository:
+            self.name = repository[CONF_NAME]
+        else:
+            self.name = repo.name
+
+        self.update = self._update
```

Comment:
Just name the method itself `update()`.

Reply:
Not sure what you mean by this one @amelchio?

Reply:
You will not need this assignment if you name the method `def update(self):`